internal/app/upstream/client_test.go

```diff
@@ -0,0 +1,221 @@
+package upstream_test
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/envoyproxy/xds-relay/internal/app/upstream"
+	mock "github.com/envoyproxy/xds-relay/test/mocks/upstream"
+	"github.com/stretchr/testify/assert"
+)
+
+type CallOptions = upstream.CallOptions
+
+func TestOpenStreamShouldReturnErrorForInvalidTypeUrl(t *testing.T) {
```

Comment:
I don't see a test here for when the request times out and gets a DeadlineExceeded error

Reply:
This one is the test: https://github.com/envoyproxy/xds-relay/blob/d3a64bf5d0ec08dd370804a29b820b1a858765e2/internal/app/upstream/client_test.go#L197-L219

We had decided that we don't need to bubble up the error to orchestrator and instead just close the channel.

Reply:
But how do you assert that the error was due to a timeout vs if the code changes? Normally, I would test this via a stat and assert that the timeout stat was incremented. 

Can you add a TODO to add that?

Reply:
Yeah i understand and i previously I meant to expose the error to the caller instead of just closing the channel. Since we have decided to close the channel to report all errors, we have to resort to alternate means to see which error really happens. I will add a TODO to check on a stat once stat is available.

Reply:
added todo.